.github/workflows/aws-e2e-tests-non-root.yaml

```diff
@@ -29,6 +29,7 @@ jobs:
      outputs:
        changed: ${{ steps.changes.outputs.changed }}
      steps:
+       - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```

Comment:
It looks like the checkout is only necessary on the `merge_group` trigger, not on `pull_request`. Can we filter this out with an `if`?